fuzzer_runner/src/buttercup/fuzzer_runner/server.py

```diff
@@ -0,0 +1,235 @@
+import asyncio
+import uuid
+from pathlib import Path
+from typing import Any
+
+from clusterfuzz.fuzz.engine import FuzzResult
+from fastapi import BackgroundTasks, FastAPI, HTTPException
+from pydantic import BaseModel, Field
+
+from buttercup.common.logger import setup_package_logger
+from buttercup.common.types import FuzzConfiguration
+from buttercup.fuzzer_runner.runner import Conf, Runner
+from buttercup.fuzzer_runner.settings import ServerSettings
+
+server_settings = ServerSettings()
+logger = setup_package_logger("fuzzer-runner", __name__, server_settings.log_level)
+
+# Global state
+app = FastAPI(
+    title="Buttercup Fuzzer Runner API",
+    description="HTTP API for running fuzzers and merging corpus",
+    version="0.1.0",
+)
+active_tasks: dict[str, dict[str, Any]] = {}
+
+
+class FuzzRequest(BaseModel):
+    corpus_dir: str = Field(..., description="Path to the corpus directory")
+    target_path: str = Field(..., description="Path to the target binary")
+    engine: str = Field(..., description="Fuzzing engine to use")
+    sanitizer: str = Field(..., description="Sanitizer to use")
+    timeout: int | None = Field(None, description="Timeout in seconds (overrides server default)")
+
+
+class MergeCorpusRequest(BaseModel):
+    corpus_dir: str = Field(..., description="Path to the corpus directory")
+    target_path: str = Field(..., description="Path to the target binary")
+    engine: str = Field(..., description="Fuzzing engine to use")
+    sanitizer: str = Field(..., description="Sanitizer to use")
+    output_dir: str = Field(..., description="Output directory for merged corpus")
+    timeout: int | None = Field(None, description="Timeout in seconds (overrides server default)")
+
+
+class FuzzResponse(BaseModel):
+    task_id: str = Field(..., description="Unique task identifier")
+    status: str = Field(..., description="Task status")
+    result: dict[str, Any] | None = Field(None, description="Fuzzing result if completed")
+    error: str | None = Field(None, description="Error message if failed")
+
+
+class MergeCorpusResponse(BaseModel):
+    task_id: str = Field(..., description="Unique task identifier")
+    status: str = Field(..., description="Task status")
+    error: str | None = Field(None, description="Error message if failed")
+
+
+class HealthResponse(BaseModel):
+    status: str = Field(..., description="Server health status")
+    version: str = Field(..., description="Server version")
+
+
+@app.get("/health", response_model=HealthResponse)
+async def health_check() -> HealthResponse:
+    """Health check endpoint"""
+    return HealthResponse(status="healthy", version="0.1.0")
+
+
+@app.post("/fuzz", response_model=FuzzResponse)
+async def run_fuzzer(request: FuzzRequest, background_tasks: BackgroundTasks) -> FuzzResponse:
```

Comment:
Consider implementing a helper function as most of the lines of run_fuzzer and merge_corpus are identical